src/adapter.rs

```diff
@@ -1192,6 +1223,55 @@ impl<'a> Adapter<'a> for RustdocAdapter<'a> {
                     unreachable!("project_neighbors {current_type_name} {edge_name} {parameters:?}")
                 }
             },
+            "Attribute" => match edge_name.as_ref() {
+                "content" => Box::new(data_contexts.map(move |ctx| {
+                    let neighbors: Box<dyn Iterator<Item = Self::DataToken> + 'a> =
+                        match &ctx.current_token {
+                            None => Box::new(std::iter::empty()),
+                            Some(token) => {
+                                let origin = token.origin;
+
+                                let attribute =
+                                    token.as_attribute().expect("token was not an Attribute");
+                                Box::new(std::iter::once(
+                                    origin.make_attribute_value_token(attribute.content.clone()),
+                                ))
+                            }
+                        };
+
+                    (ctx, neighbors)
+                })),
+                _ => {
+                    unreachable!("project_neighbors {current_type_name} {edge_name} {parameters:?}")
+                }
+            },
+            "AttributeValue" => match edge_name.as_ref() {
+                "argument" => Box::new(data_contexts.map(move |ctx| {
+                    let neighbors: Box<dyn Iterator<Item = Self::DataToken> + 'a> =
+                        match &ctx.current_token {
+                            None => Box::new(std::iter::empty()),
+                            Some(token) => {
+                                let origin = token.origin;
+
+                                let attr_value = token
+                                    .as_attribute_value()
+                                    .expect("token was not an AttributeValue");
+                                if let Some(arguments) = attr_value.arguments.clone() {
```

Comment:
This `.clone()` is still cloning an entire vector, even though the vector contains `Rc` values. Consider making it an `Rc<Vec<...>>` type, since we'll never change the contents of the vector?

I think this might be more important than wrapping `AttributeValue` in `Rc`, since if the vector is ref-counted, then then entirety of a non-refcounted `AttributeValue` can still be cloned in just 32 bytes: it's only 4 pointers.

Reply:
Could you describe the approach with `Rc<Vec<...>>` in more detail? I tried changing `Vec<...>` to `Rc<Vec<...>>` but then
```
if let Some(arguments) = attr_value.arguments.clone() {
    Box::new(arguments.iter().map(move |argument| {
        origin.make_attribute_value_token(argument.clone())
    }))
}
```
causes
```
error[E0597]: `arguments` does not live long enough
```
and I don't know how to fix it without cloning the vector (which might be consumed by `into_iter`) or collecting the resulting iterator (but this requires creating a new vector too...)

Reply:
After digging into this quite a bit myself, I have to admit it seems like this is not an optimization that's worth implementing right now, and we're better off not using an `Rc<Vec<...>>` and instead just cloning the `Vec` each time the edge is accessed as you had already implemented.

The "`arguments` does not live long enough" error is because the iterator type constructed by `.iter().map(...)` is borrowing from `arguments`, which is owned because of the `.clone()` but then dropped at the end of the function i.e. while its iterator is still live. We need to pass ownership of `arguments` into the iterator to avoid this problem.

As you've already found, `.into_iter()` doesn't work because we can't consume the vector underlying the `Rc`. Since this is because `.into_iter()` wants to output owned versions of the vector's elements, I was hoping we'd be able to avoid that problem by cloning the vector's elements before returning them — kind of like a `.into_iter().cloned()` operation but combined into one wrapping struct that takes ownership of the `Rc<Vec<...>>`.

However, after taking a look at how `.into_iter()` is [implemented for](https://doc.rust-lang.org/src/alloc/vec/into_iter.rs.html#145) `Vec`, it seems like that would involve a fair bit of `unsafe` code. The risk-vs-reward here is not worth going down that path, as it would add significant complexity and risk of bugs, for a relatively small perf gain in the big picture: many lints won't read attributes i.e. won't hit this code path, and most attributes only have a small number of `AttributeValue` components so the vectors would be small and relatively fast to clone.

Sorry to lead you down the wrong path here. You had the right approach already.